test/e2e/specs/login.js

```diff
@@ -0,0 +1,116 @@
+    LoginPage.email.click();
+    LoginPage.email.setValue('janedoe');
+    LoginPage.password.click();
+    LoginPage.emailFormatError.waitForVisible();
+    expect(LoginPage.emailFormatError.isVisible()).to.equal(true);
+  });
+
+  it('should show an error when the login fails (on mock server this is only when the incorrect email is provided)', () => {
+    LoginPage.open();
+    expect(LoginPage.loginFailed.isVisible()).to.equal(false);
+    LoginPage.email.setValue('janedoe@test.com');
+    LoginPage.password.setValue('test123');
+    LoginPage.login.click();
+    LoginPage.loginFailed.waitForVisible();
+    expect(LoginPage.loginFailed.isVisible()).to.equal(true);
+  });
+
+  it('should show both requirement errors when a login is attempted if no email and no password is provided', () => {
+    LoginPage.open();
+    expect(LoginPage.emailReqError.isVisible()).to.equal(false);
+    expect(LoginPage.passwordReqError.isVisible()).to.equal(false);
+    LoginPage.login.click();
+    LoginPage.emailReqError.waitForVisible();
+    LoginPage.passwordReqError.waitForVisible();
+    expect(LoginPage.emailReqError.isVisible()).to.equal(true);
+    expect(LoginPage.passwordReqError.isVisible()).to.equal(true);
+  });
+
+  it('should show only the email format error when a login is attempted if no password and incorrect form of email is provided', () => {
+    LoginPage.open();
+    expect(LoginPage.emailReqError.isVisible()).to.equal(false);
+    LoginPage.email.setValue('janedoe');
+    LoginPage.login.click();
+    LoginPage.emailFormatError.waitForVisible();
+    expect(LoginPage.emailFormatError.isVisible()).to.equal(true);
+  });
+
+  it('should show password requirement error when a login is attempted if no password but an email is provided', () => {
+    LoginPage.open();
+    expect(LoginPage.passwordReqError.isVisible()).to.equal(false);
+    LoginPage.email.setValue('janedoe@test.com');
+    LoginPage.login.click();
+    LoginPage.passwordReqError.waitForVisible();
+    expect(LoginPage.passwordReqError.isVisible()).to.equal(true);
+  });
+
+  it('should show email requirement error when a login is attempted if no email but a password is provided', () => {
+    LoginPage.open();
+    expect(LoginPage.emailReqError.isVisible()).to.equal(false);
+    LoginPage.password.setValue('test123');
+    LoginPage.login.click();
+    LoginPage.emailReqError.waitForVisible();
+    expect(LoginPage.emailReqError.isVisible()).to.equal(true);
+  });
+
+  it('should login successfully with a referrer query param', () => {
+    LoginPage.open();
+    LoginPage.email.setValue('parent1@example.com');
+    LoginPage.password.setValue('testparent1');
+    LoginPage.login.click();
+    LoginPage.openWithReferrer({ referrer: '/dashboard/tickets' });
+    expect(browser.getUrl()).to.equal('http://localhost:8080/dashboard/tickets');
+  });
+
+  it('should should login successfully without a referrer query param', () => {
+    LoginPage.open();
```

Comment:
Once you've fixed this to open with the referrer param correctly, then you should be opening with the referrer here. The idea is when you click login, you should be brought to the referrer URL. Your tests are passing right now because you're navigating to the referrer URL directly after login. Also, best to test with something other than '/'. '/dojos/ie/smithfield/smithfield-awesome-dojo' should work fine. You can even confirm that the Dojo details page loads afterwards.